client/src/components/dropdown-group.js

```diff
@@ -7,6 +7,25 @@ import MyDropdown from './my-dropdown';
 import CategoryDropdown from './category-dropdown';
 
 class DropdownGroup extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      categories: [], // we have empty array, that pass to CategoryDropdown
+    };
+    // eslint-disable-next-line no-console
+    console.log(this.state.categories);
+  }
+
+  // eslint-disable-next-line no-unused-vars
+  shouldComponentUpdate(nextProps, nextState, nextContent) {
+    if (typeof nextProps.categories !== 'undefined' && typeof this.props.categories !== 'undefined') {
+      if (nextProps.categories.length !== this.props.categories.length) {
+        this.setState({ categories: nextProps.categories });
+      }
+    }
+    return this.state.value !== nextState.value;
```

Comment:
It isn't a good idea to use `if` inside another `if`

Reply:
This is just for testing. If this works, of course I will make code better then. Thanks